src/driver/esp32_rmt.rs

```diff
@@ -1,150 +1,230 @@
-            ..Default::default()
-        };
-        esp!(unsafe { rmt_config(&rmt_cfg) })?;
-        esp!(unsafe { rmt_driver_install(channel, 0, 0) })?;
-        esp!(unsafe { rmt_translator_init(channel, Some(ws2812_rmt_adapter)) })?;
+    pub fn new<C: RmtChannel>(
+        channel: impl Peripheral<P = C> + 'd,
+        pin: impl Peripheral<P = impl OutputPin> + 'd,
+    ) -> Result<Self, Ws2812Esp32RmtDriverError> {
+        let config = TransmitConfig::new().clock_divider(1);
+        let tx = TxRmtDriver::new(channel, pin, &config)?;
 
-        let _encoder =
-            WS2812_ITEM_ENCODER.get_or_try_init(|| Ws2812Esp32RmtItemEncoder::new(channel))?;
+        let clock_hz = tx.counter_clock()?;
+        let encoder = Ws2812Esp32RmtItemEncoder::new(clock_hz)?;
 
-        Ok(Self { channel })
+        Ok(Self { tx, encoder })
     }
 
-    /// Writes pixel data from the slice to the IO pin.
+    /// Writes pixel data from a pixel-byte-pointer sequence to the IO pin.
     ///
     /// Byte count per LED pixel and channel order is not handled by this method.
-    /// The data has to be correctly laid out in the slice depending on the LED strip model.
+    /// The pixel data sequence has to be correctly laid out depending on the LED strip model.
     ///
     /// # Errors
     ///
     /// Returns an error if an RMT driver error occurred.
     ///
-    /// # Panics
+    /// # Warning
     ///
-    /// Panics if the given slice is longer than `u32::MAX`.
-    pub fn write(&mut self, pixel_data: &[u8]) -> Result<(), Ws2812Esp32RmtDriverError> {
-        let data_ptr = pixel_data.as_ptr();
-        esp!(unsafe { rmt_write_sample(self.channel, data_ptr, pixel_data.len(), true) })?;
+    /// Iteration of `pixel_sequence` happens inside an interrupt handler so beware of side-effects
+    /// that don't work in interrupt handlers.
+    /// See [esp_idf_hal::rmt::TxRmtDriver#start_iter_blocking()] for details.
+    pub fn write_ptr_iter_blocking<'a, 'b, T>(
+        &'a mut self,
+        pixel_sequence: T,
+    ) -> Result<(), Ws2812Esp32RmtDriverError>
+    where
+        'b: 'a,
+        T: Iterator<Item = &'b u8> + Send + 'b,
+    {
+        let signal = self.encoder.encode_iter(pixel_sequence.cloned());
+        self.tx.start_iter_blocking(signal)?;
+        Ok(())
+    }
+
+    /// Writes pixel data from a pixel-byte-value sequence to the IO pin.
+    ///
+    /// Byte count per LED pixel and channel order is not handled by this method.
+    /// The pixel data sequence has to be correctly laid out depending on the LED strip model.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if an RMT driver error occurred.
+    ///
+    /// # Warning
+    ///
+    /// Iteration of `pixel_sequence` happens inside an interrupt handler so beware of side-effects
+    /// that don't work in interrupt handlers.
+    /// See [esp_idf_hal::rmt::TxRmtDriver#start_iter_blocking()] for details.
+    pub fn write_val_iter_blocking<'a, 'b, T>(
```

Comment:
These method names are very unwieldy IMHO. I would personally opt for just calling it `write_blocking` The rest is inferred by the type of the arguments.